Terminal.Gui/Drivers/V2/WindowSizeMonitor.cs

```diff
@@ -25,14 +25,21 @@ public bool Poll ()
             return false;
         }
 
-        Size size = _consoleOut.GetWindowSize (_lastSize);
+        Size size = _consoleOut.GetWindowSize ();
 
         if (size != _lastSize)
         {
             Logging.Logger.LogInformation ($"Console size changes from '{_lastSize}' to {size}");
-            _outputBuffer.SetWindowSize (size.Width, size.Height);
-            _lastSize = size;
-            SizeChanging?.Invoke (this, new (size));
+            Size newSize = size;
+
+            if (_consoleOut.GetType().Name == "WindowsOutput")
+            {
+                newSize = _consoleOut.SetWindowSize (size);
```

Comment:
This resizing code looks confusing... can you explain why it is necessary to detect change in size and immediately order the output to resize itself again?

Reply:
But I recognize that isn't good as is. So, fell free to change all you want.